sklearn/ensemble/_weight_boosting.py

```diff
@@ -227,7 +216,7 @@ def staged_score(self, X, y, sample_weight=None):
         ------
         z : float
         """
-        X = self._validate_data(X)
+        X = self._check_X(X)
```

Comment:
Why not calling `validate_data` with `reset=False`?

Reply:
Eventually, this should become `validate_data` with `reset=True`. I went for the simplest change here.